lib/file-monitor.h

```diff
@@ -21,33 +23,35 @@
  *
  */
 
-#ifndef CFG_MONITOR_H
-#define CFG_MONITOR_H
+#ifndef FILE_MONITOR_H
+#define FILE_MONITOR_H
 
 #include "syslog-ng.h"
 #include <sys/stat.h>
 
-typedef struct _CfgMonitor CfgMonitor;
+typedef struct _FileMonitor FileMonitor;
 
-typedef struct _CfgMonitorEvent
+typedef struct _FileMonitorEvent
 {
   const gchar *name;
   struct stat st;
   enum
   {
-    MODIFIED
+    MODIFIED,
+    DELETED
   } event;
-} CfgMonitorEvent;
+} FileMonitorEvent;
 
-typedef void (*CfgMonitorEventCB)(const CfgMonitorEvent *event, gpointer c);
+typedef void (*FileMonitorEventCB)(const FileMonitorEvent *event, gpointer c);
 
-CfgMonitor *cfg_monitor_new(void);
-void cfg_monitor_free(CfgMonitor *self);
+FileMonitor *file_monitor_new(const gchar *file_name);
+void file_monitor_free(FileMonitor *self);
 
-void cfg_monitor_add_watch(CfgMonitor *self, CfgMonitorEventCB cb,  gpointer cb_data);
-void cfg_monitor_remove_watch(CfgMonitor *self, CfgMonitorEventCB cb, gpointer cb_data);
+void file_monitor_add_watch(FileMonitor *self, FileMonitorEventCB cb,  gpointer cb_data);
+void file_monitor_remove_watch(FileMonitor *self, FileMonitorEventCB cb, gpointer cb_data);
 
-void cfg_monitor_start(CfgMonitor *self);
-void cfg_monitor_stop(CfgMonitor *self);
+void file_monitor_start(FileMonitor *self);
+void file_monitor_start_silent(FileMonitor *self);
```

Comment:
It does not look like a "silent" version of start, it's more like whether you want an "immediate" check or not.

What do you think of `start_immediate()` or `start_and_check()`?
Don't you need a `bool` return value for the event callback? In that case `start_and_check()` could return that as well.